src/tensors/vector.jl

```diff
@@ -0,0 +1,45 @@
+containertype(::Type{T}) where T <:AbstractVector = T.name.wrapper
+
+unwrap(::Type{T}) where {T} = T
+
+
+single(::Type{Vector{S}}) where S = S[1]
```

Comment:
I would much rather use standard functions here, such as `single(T) = convert(T, [1])`, rather than asking people to implement several nonstandard constructors for their new array type.

Reply:
Also, if `single(T)` really is `convert(T, [1])`, why not just call `convert(T, [1])`? it's much clearer about the contents of the array

Reply:
Because [1] creates a Vector{T}, which we should avoid. We want to directly create a value of the given vector type.  We can make `single(T) = convert(T, [1])` the default behavior though.

Reply:
Actually, it technically already is it turns out.

Reply:
why can't we make a little Vector{T}?

Reply:
We don't want to do that on some random device because it means we leave how to do that up to Julia and lots of little Vector{T} could have consequences.

Reply:
To give some context on where I'm coming from: I want someone to be able to look at the constructor for `SparseListLevel` and be able to see that `pos` is initialized to `[1]` and `idx` is initialized to `[]`. I don't think that `single` is an intuitive representation of this. If you ask ChatGPT what the `single` function does in Julia, it says:

The single function in Julia is used to convert an array of floating-point numbers to single-precision floating-point numbers. Single-precision floating-point numbers use 32 bits to represent a number, while double-precision floating-point numbers (the default in Julia) use 64 bits.

Reply:
I don't think we ever call the constructor in question during moveto anyway